File: browser/src/types/Secret.ts
export enum SecretType {
	EMPTY,
	PASSWORD,
	CREDIT_CARD,
}

export enum FieldType {
	TEXT,
	HIDDEN,
}

export interface SecretSection {
	name: string | null;
	fields: SecretField[];
}

export interface SecretField {
	name: string;
	value: string;
	type: FieldType;
}

export default interface Secret {
	id: string;
	name: string;
	type: SecretType;
	icon: string | null;
	data: SecretSection[];
	custom: SecretSection[];
	notes: string;
	updated: [string, string];
	update: [string, string];
}

File: core/src/models/secret.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

use super::section::Section;

#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub enum SecretType {
	Empty,
	Login,
}

#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub struct Secret {
	pub id: String,
	pub r#type: SecretType,
	pub name: String,
	pub icon: Option<String>,
	pub data: Vec<Section>,
	pub custom: Vec<Section>,
	pub notes: Option<String>,
	pub updated_at: DateTime<Utc>,
	pub created_at: DateTime<Utc>,
}
